### PR TITLE
[IdentityCore] Add detailed SSO-JIT telemetry

### PR DESCRIPTION
**Pull request by juan-arias**
## Proposed changes

Add detailed SSO-JIT telemetry:

- Add error codes specific for each case
- Allow lastError telemetry to use a custom error string

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/MSIDError.h [hunk 1]
```diff
@@ -248,6 +248,30 @@ typedef NS_ENUM(NSInteger, MSIDErrorCode)
...
     MSIDErrorSSOExtensionUnexpectedError           =   -51815,
     
+    // JIT - Link - Timeout while waiting for server confirmation
+    MSIDErrorJITLinkServerConfirmationTimeout      =   -51816,
+    
+    // JIT - Link - Error while waiting for server confirmation
+    MSIDErrorJITLinkServerConfirmationError        =   -51817,
+    
+    // JIT - Link - Error while acquiring intune token
+    MSIDErrorJITLinkAcquireTokenError              =   -51818,
```
Comment (kaisong1990) [inline IdentityCore/src/MSIDError.h:258]: What is JIT refers to?

**Review by kaisong1990 — APPROVED** [review]

File: IdentityCore/src/MSIDError.h (hunk 1 above)
Comment (juan-arias) [reply IdentityCore/src/MSIDError.h:258]: It's "just in time", the name this feature got.

File: IdentityCore/src/MSIDError.h [hunk 2]
```diff
@@ -248,6 +248,30 @@ typedef NS_ENUM(NSInteger, MSIDErrorCode)
     // SSO Extension internal error
     MSIDErrorSSOExtensionUnexpectedError           =   -51815,
     
+    // JIT - Link - Timeout while waiting for server confirmation
+    MSIDErrorJITLinkServerConfirmationTimeout      =   -51816,
```
Comment (oldalton) [inline IdentityCore/src/MSIDError.h:252]: You need to map those in MSAL error converter as well

Comment (juan-arias) [reply IdentityCore/src/MSIDError.h:252]: I have updated the MSAL repository with the mapping: https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1529

**Review by oldalton — APPROVED** [review]

**Pull request merged**